app/controllers/search_controller.rb

```diff
@@ -1,5 +1,5 @@
 class SearchController < ApplicationController
   def new
-    @results = Resource.search(params[:query])
+    @results = params[:query].present? ? Resource.search(params[:query]) : []
```

Comment:
Minor: Can we make this not a ternary :) I find it easier to read and also it exposes costly conditional branching logic.